transformation-scripts/semopenalex-subfields.py

```diff
@@ -0,0 +1,303 @@
+
+start_time = time.ctime()
+print('subfields entity started to transform at: '+ start_time)
+
+with open(trig_output_file_path, "w", encoding="utf-8") as g:
+
+    #Path where the OpenAlex data for the current entity type is located
+    data_dump_input_entity_dir = f'{data_dump_input_root_dir}/data/{ENTITY_TYPE}/*'
+
+    for filename in glob.glob(os.path.join(data_dump_input_entity_dir, '*.gz')):
+        with gzip.open(filename, 'r') as f:
+            for line in f:
+                try:
+                    json_data = json.loads(line.decode('utf-8'))
+
+                    # subfield-ID
+                    subfield_id = json_data['id'].replace("https://openalex.org/subfields/", "")
+                    subfield_uri = URIRef(soa_namespace_subfields+subfield_id)
+                    subfields_graph.add((subfield_uri, RDF.type, soa_class_subfield))
+                    subfields_graph.add((subfield_uri, RDF.type, SKOS.Concept))
+
+                    #topic Scheme
+                    subfields_graph.add((subfield_uri, SKOS.inScheme, topic_scheme_uri))
+
+                    # display_name
+                    subfield_display_name = json_data['display_name']
+                    if not subfield_display_name is None:
+                        subfield_display_name = clean(subfield_display_name)
+                        subfields_graph.add((subfield_uri, SKOS.prefLabel, Literal(subfield_display_name,datatype=XSD.string)))
+
+                    # description
+                    topic_description = json_data['description']
+                    if not topic_description is None:
+                        topic_description = clean(topic_description)
+                        subfields_graph.add((subfield_uri, SKOS.note, Literal(topic_description,datatype=XSD.string)))
+
+                    # field
+                    topic_field = json_data['field']['id']
+                    if not topic_field is None:
+                        topic_field_id = topic_field.replace("https://openalex.org/", "")
+                        topic_field_uri = URIRef(soa_namespace_fields + topic_field_id)
+                        subfields_graph.add((subfield_uri, SKOS.broader, topic_field_uri))
+
+                    # domain
+
+                    # topics
+
+                    # ids (wikidata, wikipedia)
+                    subfield_wikidata = json_data.get('ids').get('wikidata')
+                    if not subfield_wikidata is None:
+                        subfield_wikidata = clean_url(subfield_wikidata)
+                        subfields_graph.add((subfield_uri, OWL.sameAs, URIRef(subfield_wikidata)))
+
+                    topic_wikipedia = json_data.get('ids').get('wikipedia')
+                    if not topic_wikipedia is None:
+                        topic_wikipedia = clean_url(topic_wikipedia)
+                        subfields_graph.add((subfield_uri, RDFS.seeAlso, Literal(topic_wikipedia, datatype=XSD.anyURI)))
```

Comment:
`topic_wikipedia` should be an IRI not a literal.

Reply:
I would not model wikipedia URLs as IRIs. I think a wikipedia webseite is a "normal" website and not an entity, see: `https://en.wikipedia.org/wiki/English%20language`

I only would use owl:sameAs to link to other entities in RDF KGs.